build_cmake/strict/clang_strict.cmake

```diff
@@ -6,45 +6,53 @@
 # See the file LICENSE for redistribution information.
 #
 
-string(APPEND clang_base_c_flags " -Weverything -Werror")
-string(APPEND clang_base_c_flags " -Wno-cast-align")
-string(APPEND clang_base_c_flags " -Wno-documentation-unknown-command")
-string(APPEND clang_base_c_flags " -Wno-format-nonliteral")
-string(APPEND clang_base_c_flags " -Wno-packed")
-string(APPEND clang_base_c_flags " -Wno-padded")
-string(APPEND clang_base_c_flags " -Wno-reserved-id-macro")
-string(APPEND clang_base_c_flags " -Wno-zero-length-array")
+list(APPEND clang_base_c_flags "-Weverything")
+list(APPEND clang_base_c_flags "-Werror")
+list(APPEND clang_base_c_flags "-Wno-cast-align")
+list(APPEND clang_base_c_flags "-Wno-documentation-unknown-command")
+list(APPEND clang_base_c_flags "-Wno-format-nonliteral")
+list(APPEND clang_base_c_flags "-Wno-packed")
+list(APPEND clang_base_c_flags "-Wno-padded")
+list(APPEND clang_base_c_flags "-Wno-reserved-id-macro")
+list(APPEND clang_base_c_flags "-Wno-zero-length-array")
 
 # We should turn on cast-qual, but not as a fatal error: see WT-2690.
 # For now, turn it off.
-# string(APPEND clang_base_c_flags "-Wno-error=cast-qual")
-string(APPEND clang_base_c_flags " -Wno-cast-qual")
+# list(APPEND clang_base_c_flags "-Wno-error=cast-qual")
```

Comment:
Good point, we technically don't. I originally ported it from autconf: https://github.com/wiredtiger/wiredtiger/blob/develop/build_posix/aclocal/strict.m4#L115
I believe the intention was to eventually allow `cast-qual` to come up as a warning (non-fatal error), instead of suppressing it entirely (as both an error and warning by using ` -Wno-cast-qual`). Though the comment here is pretty self explanatory so I've removed the commented line.